src/ChannelManager.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <unordered_map>
+#include <set>
+
+#include "StringUtils.h"
+
+
+class cClientHandle;
+class cCriticalSection;
+class cPluginLua;
+class cChannelCallback;
+
+
+// tolua_begin
+class cChannelManager
+{
+
+public:
+	typedef SharedPtr<cChannelCallback> cChannelCallbackRef;
+	typedef std::unordered_map<AString, cChannelCallbackRef> CallbackMap;
+
+	void AddClientToChannel      (cClientHandle & a_Client, const AString & a_Channel);
+	void AddClientToChannels     (cClientHandle & a_Client, const AString & a_Channels);
+	void RemoveClientFromChannel (cClientHandle & a_Client, const AString & a_Channel);
+	void RemoveClientFromChannels(cClientHandle & a_Client, const AString & a_Channels);
+	bool ClientHasChannel        (cClientHandle & a_Client, const AString & a_Channel);
```

Comment:
I'd redesign the village plugin to only have one channel, rather than one per village if this isn't a forge plugin. Otherwise can you look up the server plugin for forge?